### PR TITLE
Fix nested namespace module import issue

### PR DESCRIPTION
**Pull request by Hazhzeng**
This is related to running a function app with nested namespace library (e.g. azure.storage)
Repository is [here](https://github.com/jeffhollan/google-tensorflow-sample).

When we specialize the python worker from placeholder mode, we need to
1. invalidate the finder's cache.
2. reload the module to resolve the second level namespace.

Otherwise, the module will not be recognized.

### CODE REVIEW

File: azure/functions_worker/dispatcher.py [hunk 1]
```diff
@@ -374,6 +375,11 @@ def _serialize_exception(self, exc):
             for var in env_vars:
                 os.environ[var] = env_vars[var]
 
+            importlib.invalidate_caches()
+
+            for module in sys.modules.values():
+                importlib.reload(module)
```
Comment (elprans) [inline azure/functions_worker/dispatcher.py]: `reload()` is not safe in Python, especially when done to arbitrary modules.  Why is this needed?

Comment (Hazhzeng) [reply azure/functions_worker/dispatcher.py]: @elprans Hmm, because we are facing an issue when a customer uses **azure.storage.blob** library (ModuleNotFound).

In that library, there is a nested namespace, the first level is azure (which shares the same namespace as azure.functions), while the second level is azure.storage ([here](https://github.com/Azure/azure-storage-python/tree/master/azure-storage-blob/azure/storage)). The user libraries are placed in `/home/site/wwwroot/.python_packages/lib/python3.6/site-packages/azure`

When we do specialization in placeholder mode, the second level namespace "azure.storage" cannot be found.

The reason of this is when we first import **azure.functions** in placeholder mode, the **azure.__path__** is cached `['/usr/local/lib/python3.6/site-packages/azure']`.

We want to refresh the **azure** namespace to make it include the user's library location as well. So the expected **azure.__path__** should be `['/usr/local/lib/python3.6/site-packages/azure', '/home/site/wwwroot/.python_packages/lib/python3.6/site-packages/azure']`.

If there are better suggestions, please advise.
Much appreciated.

Comment (elprans) [reply azure/functions_worker/dispatcher.py]: You only need to reload `azure` then: `importlib.reload(sys.modules['azure'])`

File: azure/functions_worker/dispatcher.py [hunk 2]
```diff
@@ -374,6 +375,11 @@ def _serialize_exception(self, exc):
             for var in env_vars:
                 os.environ[var] = env_vars[var]
 
+            importlib.invalidate_caches()
```
Comment (elprans) [inline azure/functions_worker/dispatcher.py]: Please add a comment on why this is here (e.g to repopulate `azure.__path__` on worker specialization).

Comment (Hazhzeng) [reply azure/functions_worker/dispatcher.py]: Since we are only specializing the azure package, I don't think this line of code is necessary anymore. Thanks for picking this out.

Comment (Hazhzeng) [reply azure/functions_worker/dispatcher.py]: Moving to the main repo (Fixed nested namespace library import issue #492)
